fix warnings found on macos-travis

```
ld: warning: directory not found for option '-L../../cons'
4769/Users/travis/build/radareorg/radare2/libr/..//libr/bin/p/../format/elf/elf.c:3309:34: warning: comparison of unsigned expression >= 0 is always true [-Wtautological-compare]
4770        return bin->shstrtab && sh_name >= 0 && sh_name < bin->shstrtab_size;
4771                                ~~~~~~~ ^  ~
47721 warning generated.
4773In file included from /Users/travis/build/radareorg/radare2/libr/..//libr/bin/p/../format/elf/elf64.c:4:
4774/Users/travis/build/radareorg/radare2/libr/..//libr/bin/p/../format/elf/elf.c:3309:34: warning: comparison of unsigned expression >= 0 is always true [-Wtautological-compare]
4775        return bin->shstrtab && sh_name >= 0 && sh_name < bin->shstrtab_size;
4776                                ~~~~~~~ ^  ~
47771 warning generated.
4778/Users/travis/build/radareorg/radare2/libr/..//libr/bin/p/../format/mach0/mach0.c:3023:14: warning: comparison of unsigned expression < 0 is always false [-Wtautological-compare]
4779                if (amount < 0) {
4780                    ~~~~~~ ^ ~
47811 warning generated.
4782In file included from /Users/travis/build/radareorg/radare2/libr/..//libr/bin/p/../format/mach0/mach064.c:4:
4783/Users/travis/build/radareorg/radare2/libr/..//libr/bin/p/../format/mach0/mach0.c:3023:14: warning: comparison of unsigned expression < 0 is always false [-Wtautological-compare]
4784                if (amount < 0) {
4785                    ~~~~~~ ^ ~
47861 warning generated.
4787p/../arch/8051/8051_disas.c:64:14: warning: unused function 'r_8051_disas' [-Wunused-function]
4788static char *r_8051_disas(ut64 pc, const ut8 *buf, int len, int *olen) {
4789             ^
47901 warning generated.
4791/Users/travis/build/radareorg/radare2/libr/..//libr/anal/p/anal_arm_cs.c:1337:13: warning: comparison of unsigned expression < 0 is always false [-Wtautological-compare]
4792                if (shift < 0) {
4793                    ~~~~~ ^ ~
47941 warning generated.
4795/Users/travis/build/radareorg/radare2/libr/..//libr/anal/p/../..//asm/arch/tms320/c55x_plus/hashvector.c:97:25: warning: unused variable 'ins_hash' [-Wunused-variable]
4796static HASHCODE_ENTRY_T ins_hash[] = {
4797                        ^
47981 warning generated.
4799
4800
4801
```

--- a/libr/bin/format/elf/elf.c
+++ b/libr/bin/format/elf/elf.c
@@ -3304,9 +3304,8 @@ static bool is_section_local_sym(ELFOBJ *bin, Elf_(Sym) *sym) {
 	if (!is_shidx_valid (bin, sym->st_shndx)) {
 		return false;
 	}
-
 	Elf_(Word) sh_name = bin->shdr[sym->st_shndx].sh_name;
-	return bin->shstrtab && sh_name >= 0 && sh_name < bin->shstrtab_size;
+	return bin->shstrtab && sh_name < bin->shstrtab_size;
 }
 
 static void setsymord(ELFOBJ* eobj, ut32 ord, RBinSymbol *ptr) {

--- a/libr/bin/format/mach0/mach0.c
+++ b/libr/bin/format/mach0/mach0.c
@@ -1,4 +1,4 @@
-/* radare - LGPL - Copyright 2010-2019 - nibble, pancake */
+/* radare - LGPL - Copyright 2010-2020 - nibble, pancake */
 
 #include <stdio.h>
 #include <r_types.h>
@@ -3020,7 +3020,7 @@ RSkipList *MACH0_(get_relocs)(struct MACH0_(obj_t) *bin) {
 			return NULL;
 		}
 		ut64 amount = bind_size + lazy_size + weak_size;
-		if (amount < 0) {
+		if (amount == 0 || amount > UT32_MAX) {
 			return NULL;
 		}
 		relocs = r_skiplist_new ((RListFree) &free, (RListComparator) &reloc_comparator);

